pkgs/applications/networking/instant-messengers/pantalaimon/default.nix

```
@@ -51,6 +51,12 @@ buildPythonApplication rec {
   # darwin has difficulty communicating with server, fails some integration tests
   doCheck = !stdenv.isDarwin;
 
+  postPatch = ''
+    set -x
+    sed -i 's/"matrix-nio\[e2e\] <= 0.14"/"matrix-nio[e2e] >= 0.14, < 0.15"/' setup.py
```

"future-proof" tests will make sure the package is still valid
```suggestion
    sed -i 's/"matrix-nio\[e2e\] <= 0.14"/"matrix-nio[e2e]"/' setup.py
```

---

Hm, that feels a bit aggressive to me. Imagine: matrix-nio changes some security aspect of how it's used in a subtle, non-breaking way, and mentions it in release notes. WDYT?

---

I'm in favor of removing upper bounds.

---

> Imagine: matrix-nio changes some security aspect of how it's used in a subtle, non-breaking way, and mentions it in release notes. WDYT

Usually, only part of the API changes, and it might be a part that the downstream packages don't even use. That's why I usually prefer the "remove version bounds, but validate correctness" for python packages.

---

Okay. I appreciate the feedback and will keep it in mind. That having been said, there's a new release and they clamped the way I had. I'll bump to this release and back out my fix, should be good for now.